ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-file-answer/activityFileAnswer.component.html

```diff
@@ -1,27 +1,58 @@
 <ddp-question-prompt [block]="block"></ddp-question-prompt>
-<div class="file-upload-content">
+
+<div class="file-requirements" *ngIf="fileMaxSize || allowedFileTypes.length">
+    <div class="file-max-size" *ngIf="fileMaxSize">
+        {{'SDK.FileUpload.MaxSize' | translate : { maxSize: fileMaxSize | fileSize} }}
+    </div>
+    <div class="file-types" *ngIf="allowedFileTypes.length">
+        {{'SDK.FileUpload.AllowedFiles' | translate : { allowedFileTypes: getAllowedTypes(allowedFileTypes)} }}
+    </div>
+</div>
+
+<div class="file-upload-content" [class.readonly]="readonly">
     <input type="file" class="file-input"
+           [accept]="allowedFileTypes.join(',')"
+           (click)="fileUpload.value = null;"
            (change)="onFilesSelected($event.target.files)" #fileUpload>
 
     <div class="file-select">
-        <div class="drop-block" dropFileToUpload (filesDropped)="onFilesSelected($event)">
+        <div class="drop-block"
+             dropFileToUpload
+             (filesDropped)="!readonly && onFilesSelected($event)">
             <mat-icon class="upload-icon" color="primary">file_upload</mat-icon>
-            <span class="drop-block-text">Drag files to upload</span>
+            <span class="drop-block-text" translate>SDK.FileUpload.DragFiles</span>
         </div>
         <button mat-raised-button color="primary" class="upload-btn"
-                (click)="fileUpload.click()">Choose file
+                [disabled]="readonly"
+                (click)="fileUpload.click()">{{'SDK.FileUpload.ChooseFile' | translate}}
             <mat-icon>attach_file</mat-icon>
         </button>
 
+        <div class="selected-file">{{fileToUpload?.file.name}}</div>
+
+        <button *ngIf="!readonly" #uploadBtn
```

Comment:
We should disable or hide "upload" button if there is no file selected.

Reply:
Right. I lost it by mistake during a previous refactoring.
Returned back at b836160f17b9a3addca82f9f11b9faa17a2408b7.

Reply:
Eventually removed (see https://github.com/broadinstitute/ddp-angular/pull/581#issuecomment-852943813)